private/private.h

```diff
@@ -166,10 +166,22 @@ void _dispatch_prohibit_transition_to_multithreaded(bool prohibit);
  * SPI for CoreFoundation/Foundation ONLY
  */
 
-#define DISPATCH_COCOA_COMPAT (TARGET_OS_MAC || TARGET_OS_WIN32)
+#if (TARGET_OS_MAC || TARGET_OS_WIN32)
+#define DISPATCH_COCOA_COMPAT 1
+#elif defined(__linux__)
+#define DISPATCH_COCOA_COMPAT 1
+#else
+#define DISPATCH_COCOA_COMPAT 0
+#endif
 
 #if DISPATCH_COCOA_COMPAT
 
+#if TARGET_OS_WIN32
+typedef HANDLE dispatch_handle_t;
```

Comment:
see my comment on the general review, I like abstracting the type, but `dispatch_handle_t` is too generic for something that really is about runloops only. I suggested `dispatch_runloop_handle_t`. FWIW I'm not necessarily in love with `handle` as a term, `port` definitely feels wrong. Maybe @das will have a good idea, he's generally good at naming.

I would make it a typedef to mach_port_t on TARGET_OS_MAC too for consistency sake, it will be binary compatible.